core/Pos/Crypto/Encryption.hs

```diff
@@ -13,20 +13,19 @@ module Pos.Crypto.Encryption
 import           Crypto.Cipher.AES   (AES256)
 import           Crypto.Cipher.Types (BlockCipher (..), cipherInit, ctrCombine, nullIV)
 import           Crypto.Error        (CryptoError, eitherCryptoError)
-import           Crypto.Hash         (Blake2b_256)
+import           Crypto.Hash         (Blake2b_256, Digest, hash)
+import           Data.ByteArray      (convert)
 import qualified Data.Text.Encoding  as TE
-import           Pos.Binary.Class    (Bi, encodeStrict)
 import           Pos.Binary.Crypto   ()
-import           Pos.Crypto.Hashing  (AbstractHash, unsafeAbstractHash)
 import           Universum
 
 
 ----------------------------------------------------------------------------
 -- Hashing
 ----------------------------------------------------------------------------
 
-blake2b :: Bi a => a -> AbstractHash Blake2b_256 b
-blake2b = unsafeAbstractHash
+blake2b :: ByteString -> ByteString
+blake2b = convert @(Digest Blake2b_256) . hash
```

Comment:
@flyingleafe thanks, that explains it. We are encoding normal vend and force vend certificates on frontend, where paper vend seed is decoded on backend. That's why we didn't encounter it sooner.